bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/BKException.java

```diff
@@ -52,10 +67,33 @@ public final int getCode() {
         return this.code;
     }
 
+    /**
+     * Get code identifier name by code value.
+     *
+     * @param code the error code value
+     *
+     * @return the code identifier name
+     */
+    public static String getCodeName(int code) {
+        String name = codeNames.get(code);
+        return name != null ? name : Integer.toString(code);
+    }
+
+    /**
+     * Creates a lazy error code formatter suitable to pass to log functions.
+     *
+     * @param code the error code value
+     *
+     * @return lazy error code log formatter
+     */
+    public static Object codeLogger(int code) {
+        return new CodeLogFormatter(code);
```

Comment:
As the number of rc is limited can't we have a pool and save memory allocations? this code may be executed inside the write path or read path

Reply:
Sure! Please see d142c71